app/assets/javascripts/common/admins.js

```diff
@@ -107,6 +113,48 @@ AdminAccess.prototype = {
     }
   },
 
+  // walk down the tree and get the first node for each FG, then walk up the tree
+  // and update the counts
+  updateFacilityCount: function (element) {
+    const self = this
+    const leafNodesByFacilityGroup = this.getLeafNodesByFacilityGroup(element)
+    Object.values(leafNodesByFacilityGroup)
+      .filter(node => node.length > 0)
+      .map(node => node[0])
+      .forEach(node => self.updateParentFacilityCount(node, ACCESS_LIST_INPUT_SELECTOR))
+  },
+
+  updateParentFacilityCount: function (element, selector) {
+    const parent = element.closest(["ul"]).parentNode.querySelector(selector)
+    const children = nodeListToArray(selector, parent.closest("li").querySelector(["ul"]))
+    const { selected, notSelected } = children
+      .filter(({ name }) => name === "facilities[]")
+      .reduce((counter, item) => Object.assign(counter, item.checked
+        ? { selected: counter.selected + 1 }
+        : { notSelected: counter.notSelected + 1 }),
+        { selected: 0, notSelected: 0 })
+    const accessRatioDiv = element.closest(["ul"]).parentNode.querySelector(".access-ratio")
+    accessRatioDiv.textContent = notSelected == 0
+      ? `${selected} facilities selected`
+      : `${selected} of ${selected + notSelected} facilities selected`
+
+    if (element !== parent) {
+      this.updateParentFacilityCount(parent, selector)
+    }
+  },
+
+  getLeafNodesByFacilityGroup: function (element) {
+    const parent = element.closest("li")
+    const children = Array.from(parent.querySelectorAll(".access-item"))
+    return children
+      .filter(item => item.dataset.facilityGroupId)
+      .reduce((nodes, item) => {
+        const facilityGroupId = item.dataset.facilityGroupId
+        const itemsInGroup = nodes[facilityGroupId] ? [...nodes[facilityGroupId], item] : [item]
+        return Object.assign(nodes, { [facilityGroupId]: itemsInGroup })
+      }, {})
+  },
+
```

Comment:
The JS is starting to get very complicated and harder to manage. I'll refactor it a bit more, to get rid of the redundant code, but even then this might be a bit hard to read and understand.